modules/distribution/resources/api_templates/velocity_template.xml

```diff
@@ -125,7 +125,16 @@
         ######################## End of Macros ######################################
 
 
-<api xmlns="http://ws.apache.org/ns/synapse"  name="$!apiName" context="$!apiContext" version="$!apiVersion" transports="$!transport" version-type="context">
+<api xmlns="http://ws.apache.org/ns/synapse"
+     name="$!apiName"
+     context="$!apiContext"
+     version="$!apiVersion"
+     transports="$!transport"
+     version-type="context"
+     #if($apiType == 'WS')
+     binds-to="WebSocketInboundEndpoint"
```

Comment:
how do we differentiate ws and wss , here

Reply:
Even if it's `WSS`, the type set will be `WS`, and it falls into the same case.

Reply:
Added `SecureWebSocketEP` to `binds-to` as discussed offline.
Addressed in https://github.com/wso2/product-apim/pull/9885/commits/19ebe0528979bdc4ced4a3abf78d4186f15a41bb